### DISCUSSION TITLE
Retrieve secret information using the "find" method

### DISCUSSION THREAD

**Discussion by aslios** [sbertix/Swiftagram#142] [Q&A] [answered]
How can I retrieve secret information using the "find" method of the user who is already logged in with identifier?
let loginUser = getLastLoginUser() //Getting last login user instagram id
let key = KeychainStorage<Secret>.find(<#T##self: KeychainStorage<Secret>##KeychainStorage<Secret>#>)

**Comment by sbertix** [comment, marked as answer]
Hey @aslios, 

As usual, you need to create an instance of `KeychainStorage` first (either according to your needs or using the "default" one). It's not a `static` function in `KeychainStorage` but an actual method 😊 

Your code becomes… 

```swift
let loginUser = getLastUser() // Assuming this gives you a string with the logged in user identifier.
let key = KeychainStorage<Secret>().find(matching: loginUser)
```

**Reply by aslios** [reply]
I used like this but gives an error like "Instance member 'find' cannot be used on type 'KeychainStorage<Secret>'; did you mean to use a value of this type instead?"

**Reply by sbertix** [reply]
Yes, sorry hahaha I wrote the correct description and then didn't update your code.
